Level 1

Oleg021ovs opened the pull request:
123

Oleg021ovs committed:
movies api

Oleg021ovs committed:
README.md

Oleg021ovs committed:
cors

Oleg021ovs committed:
README.md

Oleg021ovs committed:
comment

Oleg021ovs committed:
исправил routes/movies.js, controllers/movies.js

StarchenkovYaroslav commented:
# Резюме:
## Backend

### Критерии, влияющие на работоспособность
- [x] <!--3.64--> В репозитории есть все необходимые инфраструктурные файлы:
  - [x] в проекте есть package.json;
  - [x] .editorconfig;
  - [x] .eslintrc, расширяющий конфигурацию airbnb-base, а также необходимые для работы линтера dev-зависимости;
  - [x] .gitignore.
  - [x] в .eslintrc добавлено исключение для _id. Директивы eslint-disable, eslint-disable-line и eslint-disable-next-line запрещены.
- [x] <!--3.64--> Нет ошибок линтера. При выполнении команды npx eslint . они должны отсутствовать.
- [ ] <!--3.64--> В разделе scripts файла package.json есть команды:
  - [ ] npm run start, которая запускает сервер на localhost:3000;
  **Скрипт запускается с хот релоудом.**
  - [x] npm run dev, которая запускает сервер на localhost:3000 с хот релоудом.
- [ ] <!--21.81--> Описанные роуты работают корректно:
  - [x] запрос на `GET /users/me` возвращает информацию о пользователе (email и имя);
  - [x] `PATCH /users/me` — обновляет информацию о пользователе;
  - [ ] `GET /movies` — все сохранённые пользователем фильмы;
  **Запрос возвращает все фильмы в базе, а не фильмы текущего пользователя.**
  - [x] `POST /movies` — создаёт фильм с переданными в теле данными;
  - [x] `DELETE /movies/movieId` — удаляет сохранённый фильмы по `_id`;
  - [x] `POST /signup` — создаёт пользователя с переданными в теле данными;
  - [x] `POST /signin` — возвращает JWT, если в теле запроса переданы правильные почта и пароль.
  - [ ] Если вы сохраняете JWT в куках, роут `/signout` должен удалять JWT из куки.
- [x] <!--3.64--> Пользователь не может удалить сохранённый фильм из профиля другого пользователя.
- [x] <!--3.64--> Все роуты, кроме /signin и /signup, защищены авторизацией.
- [x] <!--3.64--> Роуты пользователей и роуты фильмов описаны в отдельных файлах.
- [ ] <!--7.28--> Ошибки API обрабатываются:
  - [x] используются статус-коды ошибок: 400, 401, 403, 404, 409, 500;
  - [ ] если в запросе что-то не так, сервер возвращает статус ответа, который соответствует типу ошибки;
	- [ ] сообщение об ошибке соответсвует ее типу;
	**1. Не обработана ошибка несуществующего эндпоинта.**       
![2023-02-26_15-25-30](https://user-images.githubusercontent.com/86304001/221402526-0f901372-3322-48f7-96f1-f023c539262a.png)
        **2. Не обработана ошибка дубля email при редактировании пользователя.**
        **3. При общей ошибке возвращается HTML.**        
![2023-02-26_15-30-25](https://user-images.githubusercontent.com/86304001/221402748-128b7c63-7d8e-41ce-a046-b4a842cda88d.png)
  - [x] асинхронные обработчики завершены блоком catch;
  - [x] API не возвращает стандартных ошибок базы данных или Node.js.
- [x] <!--3.64--> Реализовано бережное хранение пароля:
  - [x] пароль хранится в виде хеша;
  - [x] API не возвращает хеш пароля клиенту.
- [ ] <!--7.28--> Валидация реализована верно:
  - [ ] Запрос валидируется перед передачей контроллеру. Тело и, где необходимо, заголовки и параметры проверяются по соответствующим схемам. Если запрос не соответствует схеме, обработка не передаётся контроллеру, и клиент получает ошибку валидации.
  **Запросы пользователей и фильмов не валидируются.**
  - [ ] Данные валидируются перед добавлением в базу.
  **1. Поле `movieId` на уровне схемы фильма должо валидироваться как число.**
  **2. Поле `name` на уровне схемы пользователя должно быть обязательным и не должно иметь значения по умолчанию.**
- [ ] <!--7.28--> В production-режиме адрес базы данных берётся из process.env.
**Адрес общий для всех режимов.**
- [x] <!--3.64--> К серверу можно обратиться по https по адресу домена, указанному в README.md. 
- [x] <!--7.28--> Правильно реализовано хранение секретного ключа для создания JWT:
  - [x] для продакшн сборки он хранится в .env файле — этот файл не добавляется в git;
  - [x] в режиме разработки (при process.env.NODE_ENV !== 'production') код запускается и работает и без .env файла.

### Хорошие практики
- [ ] <!--2.14--> Все роуты подключены в файле index.js, который находится в папке routes. Оттуда единый роут подключается в файле app.js.
**Роуты подключены отдельно.**
- [x] <!--2.14--> Асинхронные операции реализованы промисами или async/await.
- [ ] <!--2.14--> Валидация описана в отдельном модуле.
**Валидация описана в роутах.**
- [x] <!--2.14--> Настроено логгирование:
  - [x] запросы и ответы записываются в файл request.log;
  - [x] ошибки записываются в файл error.log;
  - [x] файлы логов не добавляются в репозиторий.
- [x] <!--2.14--> Ошибки обрабатываются централизованным обработчиком.
- [ ] <!--2.14--> Централизованный обработчик ошибок описан в отдельном модуле.
**Обработчик описан в `app.js`.**
- [x] <!--2.14--> API приложения располагается на домене вида: name.zone/api, а не просто name.zone

### Рекомендации
- [x] <!--0.71--> Для ошибок API созданы классы, расширяющие конструктор Error.
- [ ] <!--0.71--> Используется модуль Helmet для установки заголовков, связанных с безопасностью.
**Модуль не используется.**
- [ ] <!--0.71--> Конфигурация и константы в отдельных файлах:
  - [ ] адрес Mongo-сервера в режиме разработки вынесен в отдельный конфигурационный файл;
  **Адрес прописан в `app.js`.**
  - [ ] сообщения ответов и ошибок (константы приложения) вынесены в отдельный файл с константами.
  **Сообщения прописаны вручную.**
- [ ] <!--0.71--> Настроен rate limiter: число запросов с одного IP в единицу времени ограничено.
**Модуль не используется.**
- [ ] <!--0.71--> Rate limiter сконфигурирован в отдельном файле и импортируется в app.js.
**Модуль не используется.**
- [x] <!--0.71--> API размещено на отдельном поддомене, например api.movies-explorer.nomoreparties.co.
- [ ] <!--0.71--> Для имени базы данных используется `bitfilmsdb`.
**База названа `moviesdb`.**

## Количество баллов: 43

## Доп. Комментарии:
1. **"Ниже" уже нечего защищать.**
![2023-02-26_15-26-10](https://user-images.githubusercontent.com/86304001/221402570-9b599d8c-5b70-42d3-be32-96a15b118636.png)

Oleg021ovs committed:
исправил замечание ревью

Oleg021ovs committed:
1

Oleg021ovs committed:
опечатка

Oleg021ovs committed:
package-lock.json
package.json

Oleg021ovs committed:
k

Oleg021ovs committed:
удалил зависимости

StarchenkovYaroslav commented:
# Резюме:
## Backend

### Критерии, влияющие на работоспособность
- [ ] <!--3.64--> В репозитории есть все необходимые инфраструктурные файлы:
  - [x] в проекте есть package.json;
  - [x] .editorconfig;
  - [x] .eslintrc, расширяющий конфигурацию airbnb-base, а также необходимые для работы линтера dev-зависимости;
  - [x] .gitignore.
  - [ ] в .eslintrc добавлено исключение для _id. Директивы eslint-disable, eslint-disable-line и eslint-disable-next-line запрещены.
  **Присутствуют блокировки вида `eslint-disable`.**
- [ ] <!--3.64--> Нет ошибок линтера. При выполнении команды npx eslint . они должны отсутствовать.
**Ошибок не должно быть без блокировок.**
- [x] <!--3.64--> В разделе scripts файла package.json есть команды:
  - [x] npm run start, которая запускает сервер на localhost:3000;
  - [x] npm run dev, которая запускает сервер на localhost:3000 с хот релоудом.
- [x] <!--21.81--> Описанные роуты работают корректно:
  - [x] запрос на `GET /users/me` возвращает информацию о пользователе (email и имя);
  - [x] `PATCH /users/me` — обновляет информацию о пользователе;
  - [x] `GET /movies` — все сохранённые пользователем фильмы;
  - [x] `POST /movies` — создаёт фильм с переданными в теле данными;
  - [x] `DELETE /movies/movieId` — удаляет сохранённый фильмы по `_id`;
  - [x] `POST /signup` — создаёт пользователя с переданными в теле данными;
  - [x] `POST /signin` — возвращает JWT, если в теле запроса переданы правильные почта и пароль.
  - [ ] Если вы сохраняете JWT в куках, роут `/signout` должен удалять JWT из куки.
- [x] <!--3.64--> Пользователь не может удалить сохранённый фильм из профиля другого пользователя.
- [x] <!--3.64--> Все роуты, кроме /signin и /signup, защищены авторизацией.
- [x] <!--3.64--> Роуты пользователей и роуты фильмов описаны в отдельных файлах.
- [ ] <!--7.28--> Ошибки API обрабатываются:
  - [x] используются статус-коды ошибок: 400, 401, 403, 404, 409, 500;
  - [ ] если в запросе что-то не так, сервер возвращает статус ответа, который соответствует типу ошибки;
	- [ ] сообщение об ошибке соответсвует ее типу;
        **1. Не обработана ошибка дубля email при редактировании пользователя.**        
![2023-02-27_22-59-55](https://user-images.githubusercontent.com/86304001/221629726-e74c87d6-55b0-454d-b53d-e4ebd47fabae.png)
        **2. При общей ошибке возвращается HTML.**        
![2023-02-27_22-59-55](https://user-images.githubusercontent.com/86304001/221629850-2d1b806b-1e13-4e53-a863-8dfe07acddb4.png)
  - [x] асинхронные обработчики завершены блоком catch;
  - [x] API не возвращает стандартных ошибок базы данных или Node.js.
- [x] <!--3.64--> Реализовано бережное хранение пароля:
  - [x] пароль хранится в виде хеша;
  - [x] API не возвращает хеш пароля клиенту.
- [ ] <!--7.28--> Валидация реализована верно:
  - [ ] Запрос валидируется перед передачей контроллеру. Тело и, где необходимо, заголовки и параметры проверяются по соответствующим схемам. Если запрос не соответствует схеме, обработка не передаётся контроллеру, и клиент получает ошибку валидации.
  **1. Запросы пользователей и фильмов не валидируются. Валидация подключена после роута.**
  **2. При создании фильма все поля должны быть обязательными.**
  - [ ] Данные валидируются перед добавлением в базу.
  **Поле `name` на уровне схемы пользователя должно быть обязательным и не должно иметь значения по умолчанию.**
- [ ] <!--7.28--> В production-режиме адрес базы данных берётся из process.env.
**В production-режиме адрес базы данных берётся из константы.**
- [x] <!--3.64--> К серверу можно обратиться по https по адресу домена, указанному в README.md. 
- [ ] <!--7.28--> Правильно реализовано хранение секретного ключа для создания JWT:
  - [x] для продакшн сборки он хранится в .env файле — этот файл не добавляется в git;
  - [ ] в режиме разработки (при process.env.NODE_ENV !== 'production') код запускается и работает и без .env файла.
  **В режиме разработки значения берутся из окружения. Их там нет. Приложение не запускается.**  
![2023-02-27_22-55-05](https://user-images.githubusercontent.com/86304001/221628850-90770d00-441f-4e4a-830c-752add140c76.png)

### Хорошие практики
- [x] <!--2.14--> Все роуты подключены в файле index.js, который находится в папке routes. Оттуда единый роут подключается в файле app.js.
- [x] <!--2.14--> Асинхронные операции реализованы промисами или async/await.
- [ ] <!--2.14--> Валидация описана в отдельном модуле.
**Валидация для запросов авторизации и регистрации описана в роутах.**
- [x] <!--2.14--> Настроено логгирование:
  - [x] запросы и ответы записываются в файл request.log;
  - [x] ошибки записываются в файл error.log;
  - [x] файлы логов не добавляются в репозиторий.
- [x] <!--2.14--> Ошибки обрабатываются централизованным обработчиком.
- [x] <!--2.14--> Централизованный обработчик ошибок описан в отдельном модуле.
- [x] <!--2.14--> API приложения располагается на домене вида: name.zone/api, а не просто name.zone

### Рекомендации
- [x] <!--0.71--> Для ошибок API созданы классы, расширяющие конструктор Error.
- [x] <!--0.71--> Используется модуль Helmet для установки заголовков, связанных с безопасностью.
- [ ] <!--0.71--> Конфигурация и константы в отдельных файлах:
  - [x] адрес Mongo-сервера в режиме разработки вынесен в отдельный конфигурационный файл;
  - [ ] сообщения ответов и ошибок (константы приложения) вынесены в отдельный файл с константами.
  **Сообщения прописаны вручную.**
- [ ] <!--0.71--> Настроен rate limiter: число запросов с одного IP в единицу времени ограничено.
**Модуль не используется.**
- [ ] <!--0.71--> Rate limiter сконфигурирован в отдельном файле и импортируется в app.js.
**Модуль не используется.**
- [x] <!--0.71--> API размещено на отдельном поддомене, например api.movies-explorer.nomoreparties.co.
- [x] <!--0.71--> Для имени базы данных используется `bitfilmsdb`.

## Количество баллов: 59

Oleg021ovs committed:
modified:   app.js
modified:   constans/constans.js
modified:   controllers/movies.js
modified:   controllers/users.js
modified:   middlewares/auth.js
modified:   models/users.js
modified:   routes/index.js
modified:   routes/movies.js
modified:   routes/users.js

StarchenkovYaroslav commented:
# Резюме:
## Backend

### Критерии, влияющие на работоспособность
- [x] <!--3.64--> В репозитории есть все необходимые инфраструктурные файлы:
  - [x] в проекте есть package.json;
  - [x] .editorconfig;
  - [x] .eslintrc, расширяющий конфигурацию airbnb-base, а также необходимые для работы линтера dev-зависимости;
  - [x] .gitignore.
  - [x] в .eslintrc добавлено исключение для _id. Директивы eslint-disable, eslint-disable-line и eslint-disable-next-line запрещены.
- [x] <!--3.64--> Нет ошибок линтера. При выполнении команды npx eslint . они должны отсутствовать.
- [x] <!--3.64--> В разделе scripts файла package.json есть команды:
  - [x] npm run start, которая запускает сервер на localhost:3000;
  - [x] npm run dev, которая запускает сервер на localhost:3000 с хот релоудом.
- [x] <!--21.81--> Описанные роуты работают корректно:
  - [x] запрос на `GET /users/me` возвращает информацию о пользователе (email и имя);
  - [x] `PATCH /users/me` — обновляет информацию о пользователе;
  - [x] `GET /movies` — все сохранённые пользователем фильмы;
  - [x] `POST /movies` — создаёт фильм с переданными в теле данными;
  - [x] `DELETE /movies/movieId` — удаляет сохранённый фильмы по `_id`;
  - [x] `POST /signup` — создаёт пользователя с переданными в теле данными;
  - [x] `POST /signin` — возвращает JWT, если в теле запроса переданы правильные почта и пароль.
  - [ ] Если вы сохраняете JWT в куках, роут `/signout` должен удалять JWT из куки.
- [x] <!--3.64--> Пользователь не может удалить сохранённый фильм из профиля другого пользователя.
- [ ] <!--3.64--> Все роуты, кроме /signin и /signup, защищены авторизацией.
**Не защищен роут `'404 - Страницы не существует'`.**
- [x] <!--3.64--> Роуты пользователей и роуты фильмов описаны в отдельных файлах.
- [ ] <!--7.28--> Ошибки API обрабатываются:
  - [x] используются статус-коды ошибок: 400, 401, 403, 404, 409, 500;
  - [ ] если в запросе что-то не так, сервер возвращает статус ответа, который соответствует типу ошибки;
	- [ ] сообщение об ошибке соответсвует ее типу;
        **1. При общей ошибке возвращается HTML.**                
![2023-03-01_15-18-44](https://user-images.githubusercontent.com/86304001/222096384-14d2eb29-4400-44e7-b3ef-ee9f1b815614.png)
        **2. Неверная ошибка + сообщение. Проверка в принципе лишняя. Данные должны валидироваться до контроллера.**        
![2023-03-01_15-24-04](https://user-images.githubusercontent.com/86304001/222097788-a551001f-52fb-435b-8409-33fdb51db471.png)
  - [x] асинхронные обработчики завершены блоком catch;
  - [x] API не возвращает стандартных ошибок базы данных или Node.js.
- [x] <!--3.64--> Реализовано бережное хранение пароля:
  - [x] пароль хранится в виде хеша;
  - [x] API не возвращает хеш пароля клиенту.
- [ ] <!--7.28--> Валидация реализована верно:
  - [ ] Запрос валидируется перед передачей контроллеру. Тело и, где необходимо, заголовки и параметры проверяются по соответствующим схемам. Если запрос не соответствует схеме, обработка не передаётся контроллеру, и клиент получает ошибку валидации.
  **1. Запрос редактирования пользователя не валидируется. Валидация подключена после роута.**
  **2. При создании фильма ВСЕ поля должны быть обязательными.**
  - [ ] Данные валидируются перед добавлением в базу.
  **Поле `name` на уровне схемы пользователя НЕ должно иметь значения по умолчанию.**
- [x] <!--7.28--> В production-режиме адрес базы данных берётся из process.env.
- [x] <!--3.64--> К серверу можно обратиться по https по адресу домена, указанному в README.md. 
- [x] <!--7.28--> Правильно реализовано хранение секретного ключа для создания JWT:
  - [x] для продакшн сборки он хранится в .env файле — этот файл не добавляется в git;
  - [x] в режиме разработки (при process.env.NODE_ENV !== 'production') код запускается и работает и без .env файла.

### Хорошие практики
- [x] <!--2.14--> Все роуты подключены в файле index.js, который находится в папке routes. Оттуда единый роут подключается в файле app.js.
- [x] <!--2.14--> Асинхронные операции реализованы промисами или async/await.
- [ ] <!--2.14--> Валидация описана в отдельном модуле.
**Валидация для запросов авторизации и регистрации описана в роутах.**
- [x] <!--2.14--> Настроено логгирование:
  - [x] запросы и ответы записываются в файл request.log;
  - [x] ошибки записываются в файл error.log;
  - [x] файлы логов не добавляются в репозиторий.
- [x] <!--2.14--> Ошибки обрабатываются централизованным обработчиком.
- [x] <!--2.14--> Централизованный обработчик ошибок описан в отдельном модуле.
- [x] <!--2.14--> API приложения располагается на домене вида: name.zone/api, а не просто name.zone

### Рекомендации
- [x] <!--0.71--> Для ошибок API созданы классы, расширяющие конструктор Error.
- [x] <!--0.71--> Используется модуль Helmet для установки заголовков, связанных с безопасностью.
- [ ] <!--0.71--> Конфигурация и константы в отдельных файлах:
  - [x] адрес Mongo-сервера в режиме разработки вынесен в отдельный конфигурационный файл;
  - [ ] сообщения ответов и ошибок (константы приложения) вынесены в отдельный файл с константами.
  **Сообщения прописаны вручную.**
- [ ] <!--0.71--> Настроен rate limiter: число запросов с одного IP в единицу времени ограничено.
**Модуль не используется.**
- [ ] <!--0.71--> Rate limiter сконфигурирован в отдельном файле и импортируется в app.js.
**Модуль не используется.**
- [x] <!--0.71--> API размещено на отдельном поддомене, например api.movies-explorer.nomoreparties.co.
- [x] <!--0.71--> Для имени базы данных используется `bitfilmsdb`.

## Количество баллов: 78

Oleg021ovs committed:
modified:   app.js
	modified:   controllers/movies.js
	modified:   controllers/users.js
	new file:   middlewares/rateLimit.js
	modified:   models/movies.js
	modified:   models/users.js
	modified:   package-lock.json
	modified:   package.json
	modified:   routes/index.js
	modified:   routes/users.js
	modified:   validation/moviesValidation.js
	new file:   validation/registValidation.js

Oleg021ovs committed:
modified:   middlewares/errorHandler.js

StarchenkovYaroslav commented:
# Резюме:
## Backend

### Критерии, влияющие на работоспособность
- [x] <!--3.64--> В репозитории есть все необходимые инфраструктурные файлы:
  - [x] в проекте есть package.json;
  - [x] .editorconfig;
  - [x] .eslintrc, расширяющий конфигурацию airbnb-base, а также необходимые для работы линтера dev-зависимости;
  - [x] .gitignore.
  - [x] в .eslintrc добавлено исключение для _id. Директивы eslint-disable, eslint-disable-line и eslint-disable-next-line запрещены.
- [x] <!--3.64--> Нет ошибок линтера. При выполнении команды npx eslint . они должны отсутствовать.
- [x] <!--3.64--> В разделе scripts файла package.json есть команды:
  - [x] npm run start, которая запускает сервер на localhost:3000;
  - [x] npm run dev, которая запускает сервер на localhost:3000 с хот релоудом.
- [x] <!--21.81--> Описанные роуты работают корректно:
  - [x] запрос на `GET /users/me` возвращает информацию о пользователе (email и имя);
  - [x] `PATCH /users/me` — обновляет информацию о пользователе;
  - [x] `GET /movies` — все сохранённые пользователем фильмы;
  - [x] `POST /movies` — создаёт фильм с переданными в теле данными;
  - [x] `DELETE /movies/movieId` — удаляет сохранённый фильмы по `_id`;
  - [x] `POST /signup` — создаёт пользователя с переданными в теле данными;
  - [x] `POST /signin` — возвращает JWT, если в теле запроса переданы правильные почта и пароль.
  - [ ] Если вы сохраняете JWT в куках, роут `/signout` должен удалять JWT из куки.
- [x] <!--3.64--> Пользователь не может удалить сохранённый фильм из профиля другого пользователя.
- [x] <!--3.64--> Все роуты, кроме /signin и /signup, защищены авторизацией.
- [x] <!--3.64--> Роуты пользователей и роуты фильмов описаны в отдельных файлах.
- [x] <!--7.28--> Ошибки API обрабатываются:
  - [x] используются статус-коды ошибок: 400, 401, 403, 404, 409, 500;
  - [x] если в запросе что-то не так, сервер возвращает статус ответа, который соответствует типу ошибки;
	- [x] сообщение об ошибке соответсвует ее типу;
  - [x] асинхронные обработчики завершены блоком catch;
  - [x] API не возвращает стандартных ошибок базы данных или Node.js.
- [x] <!--3.64--> Реализовано бережное хранение пароля:
  - [x] пароль хранится в виде хеша;
  - [x] API не возвращает хеш пароля клиенту.
- [ ] <!--7.28--> Валидация реализована верно:
  - [x] Запрос валидируется перед передачей контроллеру. Тело и, где необходимо, заголовки и параметры проверяются по соответствующим схемам. Если запрос не соответствует схеме, обработка не передаётся контроллеру, и клиент получает ошибку валидации.
  - [ ] Данные валидируются перед добавлением в базу.
  **Поля `image`, `trailerLink`, `thumbnail` должны валидироваться на уровне схемы как ссылки.**
- [x] <!--7.28--> В production-режиме адрес базы данных берётся из process.env.
- [x] <!--3.64--> К серверу можно обратиться по https по адресу домена, указанному в README.md. 
- [x] <!--7.28--> Правильно реализовано хранение секретного ключа для создания JWT:
  - [x] для продакшн сборки он хранится в .env файле — этот файл не добавляется в git;
  - [x] в режиме разработки (при process.env.NODE_ENV !== 'production') код запускается и работает и без .env файла.

### Хорошие практики
- [x] <!--2.14--> Все роуты подключены в файле index.js, который находится в папке routes. Оттуда единый роут подключается в файле app.js.
- [x] <!--2.14--> Асинхронные операции реализованы промисами или async/await.
- [ ] <!--2.14--> Валидация описана в отдельном модуле.
**Валидация для запросов авторизации и регистрации не подключена к роутам.**
- [ ] <!--2.14--> Настроено логгирование:
  - [ ] запросы и ответы записываются в файл request.log;
  **Логгер запросов подключен после лимитера. Запросы, превышающие лимит, не попадают в лог.**
  - [x] ошибки записываются в файл error.log;
  - [x] файлы логов не добавляются в репозиторий.
- [x] <!--2.14--> Ошибки обрабатываются централизованным обработчиком.
- [x] <!--2.14--> Централизованный обработчик ошибок описан в отдельном модуле.
- [x] <!--2.14--> API приложения располагается на домене вида: name.zone/api, а не просто name.zone

### Рекомендации
- [x] <!--0.71--> Для ошибок API созданы классы, расширяющие конструктор Error.
- [x] <!--0.71--> Используется модуль Helmet для установки заголовков, связанных с безопасностью.
- [ ] <!--0.71--> Конфигурация и константы в отдельных файлах:
  - [x] адрес Mongo-сервера в режиме разработки вынесен в отдельный конфигурационный файл;
  - [ ] сообщения ответов и ошибок (константы приложения) вынесены в отдельный файл с константами.
  **Сообщения прописаны вручную.**
- [x] <!--0.71--> Настроен rate limiter: число запросов с одного IP в единицу времени ограничено.
- [x] <!--0.71--> Rate limiter сконфигурирован в отдельном файле и импортируется в app.js.
- [x] <!--0.71--> API размещено на отдельном поддомене, например api.movies-explorer.nomoreparties.co.
- [x] <!--0.71--> Для имени базы данных используется `bitfilmsdb`.

## Количество баллов: 88
